### ISSUE TITLE
What model does --save_model give?

### ISSUE THREAD

**Issue by KimberleyJensen** [facebookresearch/demucs#190]
Hello please can you tell me what model the --save_model gives from the checkpoint?

Does it give the model that had the best validation loss even if that was from many epochs ago? because i saw that the checkpoint is updated after every epoch.

One more question, is demucs only using 5 songs to validate on? because i have 30 songs in the valid folder 

![valid](https://user-images.githubusercontent.com/84926118/123795320-ce915300-d8db-11eb-8c9e-f819acf2af16.png)

Thank you for any help !

**Comment by adefossez** [comment]
Yes, `--save_model` will keep the state with the best validation loss.  The checkpoint is updated every epoch in order to resume interrupted training.

For the valid, you are probably using multi gpu training no?  In that case the tracks are splitted on each gpu, and what you see on the terminal is the progress for the first gpu.

**Comment by KimberleyJensen** [comment]
Thank you
